packages/graphql-parse-resolve-info/src/index.ts

```diff
@@ -37,6 +37,12 @@ export interface ResolveTree {
   fieldsByTypeName: FieldsByTypeName;
 }
 
+export function isResolveTree(
+  value: ResolveTree | FieldsByTypeName | null | undefined
+): value is ResolveTree {
+  return typeof value?.name === "string" && Boolean(value.fieldsByTypeName);
```

Comment:
I'm not sure this is safe; depending on how it's used it could be tricked by a query like:

```
{
  field {
    name: id # `ID` type is always a string
    fieldsByTypeName: __typename
  }
}
```